homeassistant/components/husqvarna_automower/lawn_mower.py

```diff
@@ -0,0 +1,128 @@
+from .entity import AutomowerBaseEntity
+
+SUPPORT_STATE_SERVICES = (
+    LawnMowerEntityFeature.DOCK
+    | LawnMowerEntityFeature.PAUSE
+    | LawnMowerEntityFeature.START_MOWING
+)
+
+DOCKED_ACTIVITIES = (MowerActivities.PARKED_IN_CS, MowerActivities.CHARGING)
+ERROR_ACTIVITIES = (
+    MowerActivities.STOPPED_IN_GARDEN,
+    MowerActivities.UNKNOWN,
+    MowerActivities.NOT_APPLICABLE,
+)
+ERROR_STATES = [
+    MowerStates.FATAL_ERROR,
+    MowerStates.ERROR,
+    MowerStates.ERROR_AT_POWER_UP,
+    MowerStates.NOT_APPLICABLE,
+    MowerStates.UNKNOWN,
+    MowerStates.STOPPED,
+    MowerStates.OFF,
+]
+MOWING_ACTIVITIES = (
+    MowerActivities.MOWING,
+    MowerActivities.LEAVING,
+    MowerActivities.GOING_HOME,
+)
+PAUSED_STATES = [
+    MowerStates.PAUSED,
+    MowerStates.WAIT_UPDATING,
+    MowerStates.WAIT_POWER_UP,
+]
+
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up lawn mower platform."""
+    coordinator: AutomowerDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
+    _LOGGER.debug("coordinator.data: %s", coordinator.data)
+    async_add_entities(
+        AutomowerLawnMowerEntity(mower_id, coordinator) for mower_id in coordinator.data
+    )
+
+
+class AutomowerLawnMowerEntity(LawnMowerEntity, AutomowerBaseEntity):
+    """Defining each mower Entity."""
+
+    _attr_name = None
+    _attr_supported_features = SUPPORT_STATE_SERVICES
+
+    def __init__(
+        self,
+        mower_id: str,
+        coordinator: AutomowerDataUpdateCoordinator,
+    ) -> None:
+        """Set up HusqvarnaAutomowerEntity."""
+        super().__init__(mower_id, coordinator)
+        self._attr_unique_id = f"{self.mower_id}_lawn_mower"
+
+    @property
+    def available(self) -> bool:
+        """Return True if the device is available."""
+        return self.mower_attributes.metadata.connected
+
+    @property
+    def activity(self) -> LawnMowerActivity:
+        """Return the state of the mower."""
+        mower_attributes = self.mower_attributes
+        if mower_attributes.mower.state in PAUSED_STATES:
+            return LawnMowerActivity.PAUSED
+        if mower_attributes.mower.activity in MOWING_ACTIVITIES:
+            return LawnMowerActivity.MOWING
+        if (mower_attributes.mower.state == "RESTRICTED") or (
+            mower_attributes.mower.activity in DOCKED_ACTIVITIES
+        ):
+            return LawnMowerActivity.DOCKED
+        return LawnMowerActivity.ERROR
+
+    async def async_start_mowing(self) -> None:
+        """Resume schedule."""
+        try:
+            await self.coordinator.api.resume_schedule(self.mower_id)
+        except ApiException as exception:
+            raise HomeAssistantError(
+                f"Command couldn't be sent to the command que: {exception}"
+            ) from exception
+        # await self.coordinator._async_update_data()
```

Comment:
I've missed to delete that line, from a test. Actually it works pretty good, to just send the command and wait for the websocket to send an update. I saw a lot of integration refreshing the coordinator after sending a command. What's the preferred way in HomeAssistant?

Reply:
_It depends._

I think it depends on the integration, since your integration is a push integration, just waiting for a push makes sense. (Unless that push takes 5 minutes or something like that).

If you think it can just work on push and that gives a good user experience, go for it!

Reply:
The push feedback come really fast. I leave it like that.